lib/request.js

```diff
@@ -9,6 +12,11 @@ module.exports = (options) => {
     return new Promise((resolve, reject) => {
         const chunks = [];
         const request = httpModule.request(options, (response) => {
+            response.on('error', reject);
+            if (response.headers && GZIP_ENCODINGS.indexOf(response.headers['content-encoding']) >= 0) {
```

Comment:
It's probably just a personal preference, but I feel that this is more readable (and faster 😉 ):

```js
const encoding = response.headers && response.headers['content-encoding'];
if (encoding === 'gzip' || encoding === 'deflate') {
 ...
}
```